phoenicis-apps/src/main/java/org/phoenicis/apps/MultipleRepository.java

```diff
@@ -54,4 +55,51 @@
 	public void onDelete() {
 		Arrays.stream(repositories).forEach(Repository::onDelete);
 	}
+
+	public void moveRepository(Repository repository, int toIndex) {
+		List<Repository> newRepositories = Arrays.asList(this.repositories);
+
+		int oldIndex = newRepositories.indexOf(repository);
+
+		if (oldIndex >= 0 && toIndex >= 0 && toIndex < newRepositories.size()) {
+			Collections.swap(newRepositories, oldIndex, toIndex);
+
+			this.repositories = newRepositories.toArray(new Repository[0]);
+		}
+	}
+
+	public void addRepository(Repository repository) {
```

Comment:
This is a general question: Do we want objects do something or other classes do something with objects?